### PR TITLE
Feaure/fetching data in lectures page

### PR DESCRIPTION
**Pull request by samigeorgiev**

### CODE REVIEW

File: client/src/components/UI/Input.jsx [hunk 1]
```diff
@@ -35,6 +35,17 @@ const Input = props => {
...
+                <Form.Input
+                    onChange={event =>
+                        props.onChange(event, { value: event.target.files[0] })
+                    }
+                    error={props.error}
+                    {...props.elementConfig}
+                />
+            );
+            break;
         default:
```
Comment (toniyanchev) [inline client/src/components/UI/Input.jsx:49]: can u just define it before cases and get rid of default

Comment (samigeorgiev) [reply client/src/components/UI/Input.jsx:49]: linters insist on having default case in switch

File: client/src/components/lecture/LectureEditEntry.jsx [hunk 2]
```diff
@@ -29,6 +29,7 @@ const LectureEditEntry = props => {
                     defaultValue={props.initialName}
                     transparent
                     size="huge"
+                    style={{ width: '20.75rem' }}
```
Comment (toniyanchev) [inline client/src/components/lecture/LectureEditEntry.jsx:32]: whats the difference between size and style's properties?

File: client/src/components/UI/Input.jsx (hunk 1 above)
Comment (toniyanchev) [reply client/src/components/UI/Input.jsx:49]: ok

File: client/src/components/lecture/LectureEditEntry.jsx (hunk 2 above)
Comment (samigeorgiev) [reply client/src/components/lecture/LectureEditEntry.jsx:32]: size is a component property which comes from the component library but style is direct way to inject CSS to the underlying html tag

Comment (samigeorgiev) [reply client/src/components/lecture/LectureEditEntry.jsx:32]: also size prop controls not just width but the height and font-size too

File: client/src/components/lecture/LectureEditList.jsx [hunk 3]
```diff
@@ -9,34 +22,62 @@ const LectureEditList = props => {
...
+                error
+                header="Error occurred"
+                content={state.error.message}
+            />
+        );
+    }
+    if (state.isLoading) {
+        modalContent = (
+            <Dimmer active>
+                <Loader />
```
Comment (toniyanchev) [inline client/src/components/lecture/LectureEditList.jsx:67]: what is Loader

File: client/src/components/lecture/LectureList.jsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from 'react';
+
+import { useHistory } from 'react-router-dom';
+import { List, Message, Segment } from 'semantic-ui-react';
+
+import useResourceManagementControl from '../../hooks/resourceManagementControl';
+
+import {
+    GetAllLecturesRequest as GetLecturesRequest //
```
Comment (toniyanchev) [inline client/src/components/lecture/LectureList.jsx:9]: getLecture{s} ?

File: client/src/components/lecture/LectureList.jsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                                    lecture.getId() !== props.currentLecture
+                                }
+                            />
+                            <List.Content>
+                                <List.Header
+                                    content={lecture.getName()}
+                                    onClick={() =>
+                                        lectureClickHandler(lecture.getId())
+                                    }
+                                    as="a"
```
Comment (toniyanchev) [inline client/src/components/lecture/LectureList.jsx:69]: what is a

File: client/src/components/lecture/LectureList.jsx (hunk 4 above)
Comment (toniyanchev) [reply client/src/components/lecture/LectureList.jsx:9]: NVM I thought u import getLecture frfom rm control...

File: client/src/pages/Lectures.jsx [hunk 6]
```diff
@@ -1,111 +1,68 @@
...
-                        </List.Item>
-                    ))}
-                </List>
             </Segment>
-        </>
+        );
+    }
+
+    return state.isLoading ? (
+        <Loader inline="centered" active style={{ margin: '1rem auto' }} />
```
Comment (toniyanchev) [inline client/src/pages/Lectures.jsx:63]: better visualization if on new lines?

File: proxy/envoy.yaml [hunk 7]
```diff
@@ -154,4 +154,4 @@ static_resources:
       type: logical_dns
       http2_protocol_options: {}
       lb_policy: round_robin
-      hosts: [{ socket_address: { address: resource-management-control, port_value: 9000 }
+      hosts: [{ socket_address: { address: resource-management-control, port_value: 9000} }]
```
Comment (toniyanchev) [inline proxy/envoy.yaml:157]: here as well

File: client/src/components/lecture/LectureEditEntry.jsx (hunk 2 above)
Comment (toniyanchev) [reply client/src/components/lecture/LectureEditEntry.jsx:32]: so why u resize it twice

Comment (samigeorgiev) [reply client/src/components/lecture/LectureEditEntry.jsx:32]: There isn't size which fits my needs so I choose size and fix only the width

File: client/src/components/lecture/LectureEditList.jsx (hunk 3 above)
Comment (samigeorgiev) [reply client/src/components/lecture/LectureEditList.jsx:67]: Spinner

File: client/src/components/lecture/LectureList.jsx (hunk 5 above)
Comment (samigeorgiev) [reply client/src/components/lecture/LectureList.jsx:69]: a as anchor

File: client/src/pages/Lectures.jsx (hunk 6 above)
Comment (samigeorgiev) [reply client/src/pages/Lectures.jsx:63]: Prettier has done it

File: proxy/envoy.yaml (hunk 7 above)
Comment (samigeorgiev) [reply proxy/envoy.yaml:157]: I use prettier

File: client/src/components/lecture/LectureEditEntry.jsx (hunk 2 above)
Comment (toniyanchev) [reply client/src/components/lecture/LectureEditEntry.jsx:32]: ок

File: client/src/pages/Lectures.jsx (hunk 6 above)
Comment (toniyanchev) [reply client/src/pages/Lectures.jsx:63]: okay

File: proxy/envoy.yaml (hunk 7 above)
Comment (toniyanchev) [reply proxy/envoy.yaml:157]: ok

**Pull request merged**